initrd/bin/oem-factory-reset

```diff
@@ -437,343 +833,444 @@ usb_security_token_capabilities_check
-	read -n 1 prompt_output
-	echo
-	if [ "$prompt_output" == "y" \
-			-o "$prompt_output" == "Y" ];then
-		test_luks_current_disk_recovery_key_passphrase
-		luks_new_Disk_Recovery_Key_desired=1
-		echo -e "\n"
-	fi
-
-	# Adapt message to be given to user in terms of security components that will be applied.
-	if [ -n "$luks_new_Disk_Recovery_Key_passphrase_desired" -o -n "$luks_new_Disk_Recovery_Key_passphrase" ]; then
-		CUSTOM_PASS_AFFECTED_COMPONENTS="LUKS Disk Recovery Key passphrase"
-	fi
-	if [ "$CONFIG_TPM" = "y" ]; then
-		CUSTOM_PASS_AFFECTED_COMPONENTS="$CUSTOM_PASS_AFFECTED_COMPONENTS
-TPM Ownership password"
-	fi
-	CUSTOM_PASS_AFFECTED_COMPONENTS="$CUSTOM_PASS_AFFECTED_COMPONENTS
-GPG Admin PIN
-GPG User PIN"
-
-	# Inform user of security components affected for the following prompts
-	echo -e "The following security components will be provisioned with defaults or chosen PINs/passwords:
+    #Give general guidance to user on how to answer prompts
+    echo -e "\n **** Factory Reset / Re-Ownership Questionnaire ****\n\n"
+    echo -e "The following questionnaire will help you configure the security components of your system.\n"
+    echo -e "Each prompt requires a single letter answer: eg. (Y/n).\n"
+    echo -e "If you don't know what to answer, pressing Enter will select the default answer for that prompt: eg. Y, above.\n"
+
+    # Re-ownership of encrypted disk key, content and passphrase
+    echo -e -n "\n\nWould you like to change the current LUKS Disk Recovery Key passphrase?\n (Highly recommended if you didn't install the Operating System yourself, so that past provisioned passphrase would not permit to access content.\n  Note that without re-encrypting disk, a backed up header could be restored to access encrypted content with old passphrase) [y/N]: "
+    read -n 1 prompt_output
+    echo
+    if [ "$prompt_output" == "y" \
+        -o "$prompt_output" == "Y" ]; then
+        luks_new_Disk_Recovery_Key_passphrase_desired=1
+        echo -e "\n"
+    fi
+
+    echo -e -n "Would you like to re-encrypt LUKS encrypted container and generate new Disk Recovery key?\n (Highly recommended if you didn't install the operating system yourself: this would prevent any LUKS backed up header to be restored to access encrypted data) [y/N]: "
+    read -n 1 prompt_output
+    echo
+    if [ "$prompt_output" == "y" \
+        -o "$prompt_output" == "Y" ]; then
+        test_luks_current_disk_recovery_key_passphrase
+        luks_new_Disk_Recovery_Key_desired=1
+        echo -e "\n"
+    fi
+
+    #Prompt to ask if user wants to generate GPG key material in memory or on smartcard
+    echo -e -n "Would you like to format an encrypted USB Thumb drive to store GPG key material generated in memory?\n (Required to enable GPG authentication) [y/N]: "
```

Comment:
Is the "generated in memory" part really useful?  The goal is to store key material on a USB drive, I think it's distracting from the purpose of this option.

For developers / high-tech users / etc. we do note what's happening internally below as well.

Reply:
@JonathonHall-Purism Changed to "Would you like to format an encrypted USB Thumb drive to store GPG key material?\n (Required to enable GPG authentication) [y/N]: " under [1ce158b](https://github.com/linuxboot/heads/pull/1515/commits/1ce158b1c1f0515fab5d0a41d7eae88f03552e06)